packages/server/tests/acceptance/hbarLimiter.spec.ts

```diff
@@ -503,6 +499,186 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
+              });
+            });
+
+            it('Should successfully link all pre-configured spending plans to correct NON-BASIC EVM addresses', async () => {
+              Object.entries(expectedNonBasicEvmAddresses).forEach(([key, evmAddresses]) => {
+                evmAddresses.forEach(async (evmAddress) => {
+                  const associatedPlanByEVMAddress = await ethAddressSpendingPlanRepository.findByAddress(
+                    evmAddress,
+                    requestDetails,
+                  );
+                  expect(associatedPlanByEVMAddress.planId).to.eq(expectedNonBasicPlans[key].id);
+                  expect(associatedPlanByEVMAddress.ethAddress).to.eq(evmAddress);
+                });
+              });
+            });
+
+            it('Should successfully link all pre-configured spending plans to correct NON-BASIC IP addresses', async () => {
+              Object.entries(expectedPrivilegedIpAddresses).forEach(([key, ipAddresses]) => {
+                ipAddresses.forEach(async (ipAddress) => {
+                  const associatedPlanByIpAddress = await ipSpendingPlanRepository.findByAddress(
+                    ipAddress,
+                    requestDetails,
+                  );
+                  expect(associatedPlanByIpAddress.planId).to.eq(expectedNonBasicPlans[key].id);
+                  expect(associatedPlanByIpAddress.ipAddress).to.eq(ipAddress);
+                });
+              });
+            });
+          });
+
+          describe('PRIVILEGED Tier', () => {
+            let priviledgedEvmAccount: AliasAccount;
+            let privilegedHbarSpendingPlan: IDetailedHbarSpendingPlan;
+
+            beforeEach(async () => {
+              const result = await createAliasForNonBasicPlans(SubscriptionTier.PRIVILEGED);
+              priviledgedEvmAccount = result.aliasAccount;
+              privilegedHbarSpendingPlan = result.hbarSpendingPlan;
+            });
+
+            afterEach(async () => {
+              await hbarSpendingPlanRepository.delete(privilegedHbarSpendingPlan.id, requestDetails);
+              await ethAddressSpendingPlanRepository.delete(priviledgedEvmAccount.address, requestDetails);
+            });
+
+            it('Should successfully add spending plans', async () => {
+              const plan = await ethAddressSpendingPlanRepository.findByAddress(
+                priviledgedEvmAccount.address,
+                requestDetails,
+              );
+
+              expect(plan.ethAddress).to.eq(priviledgedEvmAccount.address);
+              expect(plan.planId).to.eq(privilegedHbarSpendingPlan.id);
+
+              const spendingPlan = await hbarSpendingPlanRepository.findByIdWithDetails(plan.planId, requestDetails);
+              expect(spendingPlan.active).to.be.true;
+              expect(spendingPlan.amountSpent).to.eq(0);
+              expect(spendingPlan.subscriptionTier).to.eq(SubscriptionTier.PRIVILEGED);
+            });
```

Comment:
QQ: What's the point of this test? We are not doing anything here other than verifying the test preparations.

Reply:
This test ensures that newly created accounts are successfully loaded into the cache. However, this logic can be moved into `createAliasForNonBasicPlans` to ensure the plans are properly populated. Updated.